### PR TITLE
MAYA-114724/MAYA-114039 - Progress bar for import/export/push/pull

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-114039: On import/pull I'd like to see a progress bar

* New progress bar utility file with classes: ProgressBarScope/ProgressBarLoopScope.
* Use these new classes in read/write job import/export translator, prim updater manager and shadingModeExporter to display a progress bar for the various workflows.
* Track the number of steps advanced and output message if it doesn't equal number set.
* Added env var "MAYAUSD_ENABLE_PROGRESSBAR" (default true) which can be used to disable progress bar display.

### CODE REVIEW

File: lib/mayaUsd/fileio/jobs/readJob.cpp [hunk 1]
```diff
@@ -101,6 +102,11 @@ UsdMaya_ReadJob::~UsdMaya_ReadJob() { }
 
 bool UsdMaya_ReadJob::Read(std::vector<MDagPath>* addedDagPaths)
 {
+    // When we are called from PrimUpdaterManager we should already have
+    // a computation scope. If we are called from elsewhere don't show any
+    // progress bar here.
+    MayaUsd::ProgressBarScope progressBar(16);
```
Comment (seando-adsk) [inline lib/mayaUsd/fileio/jobs/readJob.cpp:108]: Using this constructor (just a steps number) will add to an existing (if one exists) progress bar. If no progress bar exists, this does nothing.

File: lib/mayaUsd/fileio/jobs/readJob.cpp [hunk 2]
```diff
@@ -510,6 +534,9 @@ bool UsdMaya_ReadJob::_DoImport(UsdPrimRange& rootRange, const UsdPrim& usdRootP
             ? UsdPrimRange::PreAndPostVisit(rootPrim)
             : UsdPrimRange::PreAndPostVisit(
                 rootPrim, UsdTraverseInstanceProxies(UsdPrimAllPrimsPredicate));
+
+        const int loopSize = std::distance(range.begin(), range.end());
+        MayaUsd::ProgressBarLoopScope instanceLoop(loopSize);
```
Comment (seando-adsk) [inline lib/mayaUsd/fileio/jobs/readJob.cpp:539]: How to add steps for a loop

File: plugin/adsk/plugin/exportTranslator.cpp [hunk 3]
```diff
@@ -56,6 +57,10 @@ MStatus UsdMayaExportTranslator::writer(
 
     std::string fileName(file.fullName().asChar(), file.fullName().length());
 
+    MString progStatus, mStrFileName(fileName.c_str());
+    progStatus.format("Writing ^1s", mStrFileName);
+    MayaUsd::ProgressBarScope progressBar(3, progStatus);
```
Comment (seando-adsk) [inline plugin/adsk/plugin/exportTranslator.cpp:62]: Example where a new progress bar will be created showing the string given.

File: plugin/adsk/scripts/mayaUSDRegisterStrings.mel [hunk 4]
```diff
@@ -212,7 +212,6 @@ global proc mayaUSDRegisterStrings()
     register("kImportUSDZTxtAnn", "When a .usdz file is chosen and this is selected, .usdz texture files are imported and copied to new files on disk. To locate these files, navigate to the current Maya workspace /sourceimages directory.");
     register("kImportUSDZTxtLbl", "USDZ Texture Import: ");
     register("kImportVariantsInScopeNumLbl", "Variants Switched in Scope:");
-    register("kMayaRefMergeEdits", "Merge Maya Edits to USD");
```
Comment (seando-adsk) [inline plugin/adsk/scripts/mayaUSDRegisterStrings.mel:215]: This string is not used, was old.

File: lib/mayaUsd/utils/progressBarScope.h [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    void advance(const int steps = 1);
+
+    bool isInterruptRequested() const;
+
+private:
+    bool _created { false };
+    int  _nbSteps { 0 };
+
+    // There can be only one progress bar during a given operation.
+    static std::unique_ptr<MComputation> progBar;
```
Comment (seando-adsk) [inline lib/mayaUsd/utils/progressBarScope.h:78]: There can only be one progress bar at a time.

File: lib/mayaUsd/utils/progressBarScope.cpp [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+ProgressBarScope::ProgressBarScope(
+    bool           showProgress,
+    bool           interruptible,
+    const int      nbSteps,
+    const MString& progressStr)
+{
+    // Allow user (or QA) to disable the display of the progress bar.
+    // Default display value is ON.
+    bool createProgressBar = TfGetenvBool("MAYAUSD_ENABLE_PROGRESSBAR", true);
```
Comment (seando-adsk) [inline lib/mayaUsd/utils/progressBarScope.cpp:46]: Check env var to disable progress bar.

**Review by pierrebai-adsk — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**